tests/std/tests/P1522R1_difference_type/test.cpp

```diff
@@ -3,26 +3,45 @@
 
 #include <__msvc_int128.hpp>
 #include <cassert>
-#include <compare>
-#include <concepts>
 #include <cstdint>
 #include <iterator>
 #include <limits>
 #include <type_traits>
 
+#if _HAS_CXX20
+#include <compare>
+
+#ifndef __EDG__ // TRANSITION, GH-395
+#include <concepts>
+#endif // __EDG__
+#endif // _HAS_CXX20
+
+#if _HAS_CXX20 && !defined(__EDG__) // TRANSITION, GH-395
+#define SAME_AS std::same_as
+#else // ^^^ has concepts / vvv has no concepts
+#define SAME_AS std::is_same_v
+#endif // _HAS_CXX20 && !defined(__EDG__)
+
+#define STATIC_ASSERT(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
+
 using std::_Signed128;
 using std::_Unsigned128;
+using namespace std::literals::_Int128_literals;
 
-constexpr void check_equal(const auto& x) {
+template <class I>
+constexpr void check_equal(const I& x) {
     assert(x == x);
     assert(!(x != x));
     assert(!(x < x));
     assert(!(x > x));
     assert(x <= x);
     assert(x >= x);
+#if _HAS_CXX20
     assert(x <=> x == 0);
+#endif // _HAS_CXX20
 }
 
+#if _HAS_CXX20

```

Comment:
It'd be nice if this check was also C++14, and we just have a test-internal `strong_ordering` replacement

Reply:
I failed to find that internal replacement so I re-implemented it...😿